facilitator/src/bin/facilitator.rs

```diff
@@ -743,36 +848,218 @@ fn main() -> Result<(), anyhow::Error> {
+                Err(e) => error!("\tError: {:?}", e),
+            }
+        });
+    }
+}
+
+fn get_ecies_public_key(
+    key_option: Option<&str>,
+    manifest_url: Option<&str>,
+    ingestor_name: &str,
+    locality_name: &str,
+) -> Result<PublicKey> {
+    let peer_name = &format!("{}-{}", locality_name, ingestor_name);
+    match key_option {
+        Some(key) => PublicKey::from_base64(key)
+            .map_err(|e| anyhow!("unable to create public key from base64 ecies key: {}", e)),
+        None => match manifest_url {
+            Some(manifest_url) => {
+                let manifest = SpecificManifest::from_https(manifest_url, peer_name).context(
+                    format!("unable to read SpecificManifest from {}", manifest_url),
+                )?;
+                let packet_decryption_keys = manifest
+                    .packet_decryption_keys()
+                    .context("unable to get packet decryption keys from the SpecificManifest")?;
+                let (key_identifier, packet_decryption_key) = packet_decryption_keys
+                    .into_iter()
+                    .next()
+                    .context("No packet decryption keys in manifest")?;
+
+                let public_key = packet_decryption_key.base64_public_key()?;
+
+                let public_key = PublicKey::from_base64(&public_key).map_err(|e| {
+                    anyhow!("unable to create public key from base64 ecies key: {}", e)
+                })?;
+
+                info!(
+                    "Picked packet decryption key with ID: {} - public key {:?}",
+                    key_identifier, &public_key
+                );
+
+                Ok(public_key)
+            }
+            None => panic!(
+                "Neither manifest_option or key_option were specified. This error shouldn't happen."
+            ),
+        },
+    }
+}
+
+fn get_identity_and_bucket(
+    identity: Option<&str>,
+    bucket: Option<&str>,
+    manifest_url: Option<&str>,
+    ingestor_name: &str,
+    locality_name: &str,
+) -> Result<(Option<String>, String)> {
+    let peer_name = &format!("{}-{}", locality_name, ingestor_name);
+    match bucket {
+        Some(bucket) => Ok((identity.map(String::from), String::from(bucket))),
+        None => {
+            let manifest_url =
+                manifest_url.expect("If bucket is not provided, manifest_url must be provided");
+            let manifest = SpecificManifest::from_https(manifest_url, peer_name).context(
+                format!("unable to read SpecificManifest from {}", manifest_url),
+            )?;
+
+            Ok((manifest.ingestion_identity(), manifest.ingestion_bucket()))
+        }
+    }
+}
+
+fn get_valid_batch_signing_key(
+    namespace: Option<&str>,
+    ingestor_manifest_url: Option<&str>,
+    matches: &ArgMatches,
+) -> Result<BatchSigningKey> {
+    match ingestor_manifest_url {
+        Some(own_manifest_url) => {
+            let namespace = namespace.unwrap();
```

Comment:
I unwrap when ingestor_manifest_url is provided. If it's not then it goes to try to get the BSK from `batch_signing_key_from_arg`

Reply:
Yeah but what if `namespace` is `None` and `own_manifest_url` is `Some`? Then we'll crash here. Admittedly `unwrap()` is used rather casually all over `facilitator.rs` so it's not the end of the world, but since this function returns `Result<BatchSigningKey>` already, it should be easy to return a proper error.

Reply:
That's technically incorrect use of the CLI. I'll add an error message saying namespace must be supplied when using the `own_manifest_url` method.

I'm not super concerned about crashing immediately on startup since the recovery is nearly always going to be something the user did wrong when using the tool. I'll bubble the error up for the caller to handle.